client/response.go

```diff
@@ -17,6 +17,8 @@ func (r *Response) Unmarshal(v interface{}) (err error) {
 
 	switch r.Header.Get(headers.ContentEncoding) {
 	case "gzip":
+		defer reader.Close()
+
```

Comment:
funkar rad 27 om vi unmarshallar en response utan body? Eller får vi panic?

Reply:
> The http Client and Transport guarantee that Body is always non-nil, even on responses without a body or responses with a zero-length body. 

So the close on row 27 will always work